src/sys/fuchsia/selector.rs

```diff
@@ -0,0 +1,369 @@
+use {io, Event, PollOpt, Ready, Token};
+use sys::fuchsia::{
+    epoll_event_to_ready,
+    poll_opts_to_wait_async,
+    status_to_io_err,
+    EventedFd,
+    EventedFdInner,
+};
+use magenta;
+use magenta_sys;
+use magenta::HandleBase;
+use std::collections::hash_map;
+use std::fmt;
+use std::mem;
+use std::sync::atomic::{AtomicBool, AtomicUsize, ATOMIC_USIZE_INIT, Ordering};
+use std::sync::{Arc, Mutex, Weak};
+use std::time::Duration;
+use sys;
+
+/// The kind of registration-- file descriptor or handle.
+///
+/// The last bit of a token is set to indicate the type of the registration.
+#[derive(Copy, Clone, Eq, PartialEq)]
+enum RegType {
+    Fd,
+    Handle,
+}
+
+fn key_from_token_and_type(token: Token, reg_type: RegType) -> io::Result<u64> {
+    let key = token.0 as u64;
+    let msb = 1u64 << 63;
+    if (key & msb) != 0 {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "Most-significant bit of token must remain unset."));
```

Comment:
Yeah Tokio currently assigns (AFAIK at least) a unique token per event and tokens that start counting from 0. So at least for tokio's use case we won't hit this error I believe, but it's true that for arbitrary usage of `mio` this'd cause problems (but returning an error I think is fine)